platform/api/aws/ami.go

```diff
@@ -28,22 +28,9 @@ type releaseAMIs struct {
 	}
 }
 
-func (a *API) AMI() (resolvedAMI string, resolveAMIErr error) {
-	a.opts.resolvedAMILock.Lock()
-	defer a.opts.resolvedAMILock.Unlock()
-
-	// Use cache if we can
-	if a.opts.ResolvedAMI != "" {
-		return a.opts.ResolvedAMI, nil
-	}
-
-	// If successfully resolved, cache the AMI id for next call.
-	// Decide based on the named returns
-	defer func() {
-		if resolveAMIErr == nil {
-			a.opts.ResolvedAMI = resolvedAMI
-		}
-	}()
+func (a *API) AMI() (string, error) {
+	a.amiResolutionLock.Lock()
+	defer a.amiResolutionLock.Unlock()
```

Comment:
I hadn't anticipated that you'd update `a.opts.AMI` in place.  That is indeed the typical approach in mantle, but it now races against other API objects sharing the Options struct (e.g. in `kola run --parallel`).  Arguably this can be made simpler and more consistent by dropping the `AMI` method and lock (or, if we want to be correct, making the lock static or moving it back to Options) and moving resolution into `New()`.